.github/CONTRIBUTING.md

```diff
@@ -74,6 +87,40 @@ Before submitting a PR please verify that your code passes all the tests. To do
 so run the `unit-tests()` function along with any other function specified by the
 documentation in the [tests file][Tests file].
 
+## PR management
+
+This is the optimal workflow that should be followed when managing a new PR:
+
+1. The submitter (a collaborator or an external user) submits the PR;
+2. A collaborator of the project assigns some other collaborators (or his/herself)
+  to the PR;
+3. The assigned user adds the correct tags/milestones to the PR;
+4. The assigned user performs a general evaluation of the PR: if he finds any big
+  problem (exceptionally bad code, the feature added by the PR is not wanted in the
+  project or something along these lines) he can simply refuse the PR by closing
+  it (stating the problem clearly in a comment and discussing it with the submitter);
+5. If no big problem is found the assigned user starts a review of the code by adding
+  one or more reviewers (other collaborators or him/herself) to the PR;
+6. The reviewers should check the code for errors, problems, possible optimizations,
+  incompatibilities with the current code base or with planned future developments,
+  wrong code style and typos/spelling errors and running the tests, then leave a
+  review accepting it if everything is OK or requesting changes otherwise;
+7. If the submitter is asked by the reviewer(s) to implement some changes in the
+  code he should do it before the process can continue any further.  
+  As a general rule reviewers should refrain from pushing commits to the PR branch
+  without asking explicit consent from User A beforehand.  
+  Note: it's perfectly OK for both the reviewers and the submitter to use `git rebase`
+  while working on PRs since nobody should fork a branch from an unmerged PR;
+8. If any changes have been pushed to the PR in step 7 step 6 must be repeated;
+9. Once all the reviewers have accepted the PR they can ask the submitter to rebase,
+  squash or tidy up the PR branch before merging (For example, this should be done
+  if the review/correction cycle was repeated multiple times generating lots of
+  unwanted commits that can be squashed);
+10. Once the PR is finally ready to be merged the assigned user should perform:
+    - a "fast-forward merge" (manually) if possible;
```

Comment:
I really appreciate your insights as you have tons of experience in this field and I really like learning new things: while I understand what you are saying I think that for this project having merge commits for every PR would cause more trouble than good since most of the PRs submitted during a _normal state of operation_ would be small fixes and translations (mainly 1-commit PRs). However there are times (like the last weeks) when more substantial changes are made and in these cases I agree that `--no-ff` merges would help.

To sum up, this is what I would do to get the best of the two worlds:

- if the PR consists of 1 or 2 commits then `merge -ff` or `rebase` (if `-ff` is impossible) it;
- if the PR contains more than 2 commits `merge -no-ff`;
- always use your head: if the standard approach would cause more trouble than it solves then break it and provide an alternative solution for that particolar case;

Reply:
(...just replying to your comment with an ACK, agreed that it is good to have a guideline like you just described, but be able to diverge from it on a context-by-context basis)